lib/tether/tether_linux.go

```diff
@@ -130,20 +130,27 @@ func (t *tether) childReaper() error {
 						continue
 					}
 
-					log.Debugf("Reaped process %d, return code: %d", pid, status.ExitStatus())
+					exitCode := status.ExitStatus()
+					log.Debugf("Reaped process %d, return code: %d", pid, exitCode)
 
 					session, ok := t.removeChildPid(pid)
-					log.Debugf("Remove child pid: %d ok: %t", pid, ok)
 					if ok {
+						log.Debugf("Removed child pid: %d", pid)
 						session.Lock()
-						session.ExitStatus = status.ExitStatus()
+						session.ExitStatus = exitCode
 
 						t.handleSessionExit(session)
 						session.Unlock()
-					} else {
-						// This is an adopted zombie. The Wait4 call already clean it up from the kernel
-						log.Warnf("Reaped zombie process PID %d", pid)
+						continue
+					}
+
+					ok = t.ops.HandleUtilityExit(pid, exitCode)
```

Comment:
I think fundamentally this is what my PR was missing, this along w/ the `HandleUtilityExit` method. Also all those beautiful tests.